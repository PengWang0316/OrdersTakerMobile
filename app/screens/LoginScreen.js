import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import SnackBar from '@kevinwang0316/react-native-snackbar-component';

import SafeAreaScreen from './SafeAreaScreen';
import LoginPanel from '../components/LoginPanel';
import RegisterPanel from '../components/RegisterPanel';
import Theme from '../Theme';
// import LoginScreenContext from '../contexts/LoginScreenContext';

type Props = {
  navigation: Object,
};

type States = {
  isShowSnackbar: boolean,
  snackbarMessage: string,
  snackbarBgColor: string,
  isShowRegisterPanel: boolean,
};

const AUTO_HIDDING_TIME = 1500;
const REGISTER_TEXT = 'Do Not Have An Account? Register One';
const LOGIN_TEXT = 'Have An Account Already? Go To Login';

/**
 * The component that contains login form and register form.
 */
export class LoginScreen extends Component<Props, States> {
  state = {
    isShowSnackbar: false,
    snackbarMessage: '',
    snackbarBgColor: '',
    isShowRegisterPanel: false,
  };

  /**
   * Somehow the test will fail if the Component that extends from Component does not call the super(props) manually.
   * It could relate to babel, react-native, or metro-react-native-babel-preset.
   * Will be removed if the problem disappear in the future.
   * @param {object} props has all props value
   */
  constructor(props) {
    super(props);
  }

  /**
   * Updating the states that relates to the snackbar.
   * @param {object} snackbarStates contains isShowSnackbar, snackbarMessage, and snackbarBgColor.
   * @return {null} No return.
   */
  snackbarUpdateCallback = snackbarStates => this.setState(
    snackbarStates,
    () => setTimeout(() => this.setState({ isShowSnackbar: false }), AUTO_HIDDING_TIME + 100),
  );

  /**
   * Set the isShowRegisterPanel state to the opposite value.
   * @return {null} No return.
   */
  handlePanelSwitch = () => this.setState(({ isShowRegisterPanel }) => ({ isShowRegisterPanel: !isShowRegisterPanel }));

  /**
   * Render function for the component.
   * @return {jsx} Return jsx.
   */
  render() {
    const {
      isShowSnackbar, snackbarMessage, snackbarBgColor, isShowRegisterPanel,
    } = this.state;
    const { navigation } = this.props;
    return (
      <SafeAreaScreen>
        {/* <LoginScreenContext.Provider value={{ handleSnackbarUpdate: this.snackbarUpdateCallback }}> */}
        {!isShowRegisterPanel && <LoginPanel handleSnackbarUpdate={this.snackbarUpdateCallback} navigation={navigation} />}
        {/* <Button onPress={() => navigation.goBack()} title="Go Back" /> */}
        {isShowRegisterPanel && <RegisterPanel handleSnackbarUpdate={this.snackbarUpdateCallBack} navigation={navigation} />}

        <View>
          <TouchableOpacity
            onPress={this.handlePanelSwitch}
            style={{ marginTop: 20 }}
          >
            <Text style={{ color: Theme.primary.light }}>{isShowRegisterPanel ? LOGIN_TEXT : REGISTER_TEXT}</Text>
          </TouchableOpacity>
        </View>
        <SnackBar
          visible={isShowSnackbar}
          textMessage={snackbarMessage}
          backgroundColor={snackbarBgColor}
          autoHidingTime={AUTO_HIDDING_TIME}
        />
        {/* </LoginScreenContext.Provider> */}
      </SafeAreaScreen>
    );
  }
}

export default LoginScreen;
