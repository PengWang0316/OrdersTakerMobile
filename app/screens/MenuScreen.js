import React from 'react';
import { StyleSheet, Text } from 'react-native';

import SafeAreaScreen from './SafeAreaScreen';

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    margin: 10,
  },
});

export const MenuScreen = () => (
  <SafeAreaScreen>
    <Text style={styles.welcome}>Welcome to the Menu Screen!</Text>
  </SafeAreaScreen>
);
export default MenuScreen;
