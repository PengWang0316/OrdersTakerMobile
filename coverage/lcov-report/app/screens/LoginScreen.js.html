<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/screens/LoginScreen.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/screens</a> LoginScreen.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>40/40</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>19/19</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { View, TouchableOpacity, Text } from 'react-native';
import SnackBar from '@kevinwang0316/react-native-snackbar-component';
&nbsp;
import SafeAreaScreen from './SafeAreaScreen';
import LoginPanel from '../components/LoginPanel';
import RegisterPanel from '../components/RegisterPanel';
import Theme from '../Theme';
// import LoginScreenContext from '../contexts/LoginScreenContext';
&nbsp;
// type Props = {
//   navigation: Object
// };
&nbsp;
type States = {
  isShowSnackbar: boolean,
  snackbarMessage: string,
  snackbarBgColor: string,
  isShowRegisterPanel: boolean,
};
&nbsp;
const AUTO_HIDDING_TIME = 1500;
const REGISTER_TEXT = 'Do Not Have An Account? Register One';
const LOGIN_TEXT = 'Have An Account Already? Go To Login';
&nbsp;
/**
 * The component that contains login form and register form.
 */
export class LoginScreen extends Component&lt;null, States&gt; {
  state = {
    isShowSnackbar: false,
    snackbarMessage: '',
    snackbarBgColor: '',
    isShowRegisterPanel: false,
  };
&nbsp;
  /**
   * Updating the states that relates to the snackbar.
   * @param {object} snackbarStates contains isShowSnackbar, snackbarMessage, and snackbarBgColor.
   * @return {null} No return.
   */
  snackbarUpdateCallback = snackbarStates =&gt; this.setState(
    snackbarStates,
    () =&gt; setTimeout(() =&gt; this.setState({ isShowSnackbar: false }), AUTO_HIDDING_TIME + 100),
  );
&nbsp;
  /**
   * Set the isShowRegisterPanel state to the opposite value.
   * @return {null} No return.
   */
  handlePanelSwitch = () =&gt; this.setState(({ isShowRegisterPanel }) =&gt; ({ isShowRegisterPanel: !isShowRegisterPanel }));
&nbsp;
  /**
   * Render function for the component.
   * @return {jsx} Return jsx.
   */
  render() {
    const {
      isShowSnackbar, snackbarMessage, snackbarBgColor, isShowRegisterPanel,
    } = this.state;
    return (
      &lt;SafeAreaScreen&gt;
        {/* &lt;LoginScreenContext.Provider value={{ handleSnackbarUpdate: this.snackbarUpdateCallback }}&gt; */}
        {!isShowRegisterPanel &amp;&amp; &lt;LoginPanel handleSnackbarUpdate={this.snackbarUpdateCallback} /&gt;}
        {/* &lt;Button onPress={() =&gt; navigation.goBack()} title="Go Back" /&gt; */}
        {isShowRegisterPanel &amp;&amp; &lt;RegisterPanel handleSnackbarUpdate={this.snackbarUpdateCallBack} /&gt;}
&nbsp;
        &lt;View&gt;
          &lt;TouchableOpacity
            onPress={this.handlePanelSwitch}
            style={{ marginTop: 20 }}
          &gt;
            &lt;Text style={{ color: Theme.primary.light }}&gt;{isShowRegisterPanel ? LOGIN_TEXT : REGISTER_TEXT}&lt;/Text&gt;
          &lt;/TouchableOpacity&gt;
        &lt;/View&gt;
        &lt;SnackBar
          visible={isShowSnackbar}
          textMessage={snackbarMessage}
          backgroundColor={snackbarBgColor}
          autoHidingTime={AUTO_HIDDING_TIME}
        /&gt;
        {/* &lt;/LoginScreenContext.Provider&gt; */}
      &lt;/SafeAreaScreen&gt;
    );
  }
}
&nbsp;
export default LoginScreen;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 30 2018 14:48:02 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
