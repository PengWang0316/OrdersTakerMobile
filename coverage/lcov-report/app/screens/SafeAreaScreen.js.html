<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/screens/SafeAreaScreen.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/screens</a> SafeAreaScreen.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32</td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
&nbsp;
import DarkerStatusBar from '../components/DarkerStatusBar';
import Theme from '../Theme';
&nbsp;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Theme.panelBackgroundColor,
  },
});
&nbsp;
type Props = {
  children: Object,
  style: Object,
};
/**
 * Wrap the children components to a safe area component with a customizedstatus bar.
 * @param {object} children has all component's children component.
 * @return {jsx} Return jsx for the component.
 */
export const SafeAreaScreen = ({ children, style }: Props) =&gt; (
  &lt;SafeAreaView style={[styles.container, style]}&gt;
    &lt;DarkerStatusBar /&gt;
    {children}
  &lt;/SafeAreaView&gt;
);
export default SafeAreaScreen;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Sep 29 2018 18:24:04 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
