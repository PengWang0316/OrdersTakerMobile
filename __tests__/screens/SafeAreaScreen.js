import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import { SafeAreaScreen } from '../../app/screens/SafeAreaScreen';

jest.mock('SafeAreaView', () => 'SafeAreaView');
jest.mock('Text', () => 'Text');

describe('SafeAreaScreen Test', () => {
  test('Snapshot test', () => expect(renderer.create(<SafeAreaScreen><Text>text</Text></SafeAreaScreen>).toJSON()).toMatchSnapshot());
});
