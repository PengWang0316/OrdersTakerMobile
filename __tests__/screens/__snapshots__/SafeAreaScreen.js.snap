// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeAreaScreen Test Snapshot test 1`] = `
<SafeAreaView
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text>
    text
  </Text>
</SafeAreaView>
`;
